azureeyemodule/app/iot/iot_interface.cpp

```diff
@@ -147,47 +147,66 @@ static inline std::unordered_map<std::string, std::string> parse_onvif_message(c
     // This naive format works for something as simple as this for now, but we should really consider
     // using a better serialization/deserialization mechanism.
 
-    const std::string delimiters = "-_";
+    const std::string delimiter1 = "-"; // the delimiter to seperate different feature
+    const std::string delimiter2 = "_"; // the delimiter to deperate different parameters for each feature
     std::string dup = std::string(msg);
 
     // Tokenize the string into a vector of strings
     std::vector<std::string> tokens_and_values;
     size_t pos = 0;
-    while ((pos = dup.find_first_of(delimiters)) != std::string::npos)
+    while ((pos = dup.find_first_of(delimiter1)) != std::string::npos)
     {
         std::string part = dup.substr(0, pos);
+	printf("- parsed : %s \n", part.c_str());
```

Comment:
Two things:

1. Please use util::log_debug or util::log_info (or just remove this if you don't want to log it) instead of printf directly.
2. Second, please configure your editor to use spaces instead of tabs. Boo tabs! And then replace all the tabs in this file with spaces. Thaaaaaanks